src/libfuncs.rs

```diff
@@ -435,3 +445,39 @@ fn increment_builtin_counter_by<'ctx: 'a, 'a>(
         location,
     ))
 }
+
+fn build_noop<'ctx, 'this, const N: usize, const PROCESS_BUILTINS: bool>(
+    context: &'ctx Context,
+    registry: &ProgramRegistry<CoreType, CoreLibfunc>,
+    entry: &'this Block<'ctx>,
+    location: Location<'ctx>,
+    helper: &LibfuncHelper<'ctx, 'this>,
+    _metadata: &mut MetadataStorage,
+    param_signatures: &[ParamSignature],
+) -> Result<()> {
+    let mut params = Vec::with_capacity(N);
+
+    #[allow(clippy::needless_range_loop)]
+    for i in 0..N {
+        let param_ty = registry.get_type(&param_signatures[i].ty)?;
+        let mut param_val = entry.argument(i)?.into();
+
+        if PROCESS_BUILTINS
```

Comment:
Couldn't we iterate over param_signatures instead of using a const generic?

Reply:
While we technically could, in this case we depend on the constant `N`. It should never happen, but if the number of arguments was smaller than the constant then the number of values to return would be smaller than the constant `N`.

Reply:
Aaah. Is there a situation where the number of return values is smaller than the number of arguments?

Reply:
None. This can only happen on:
  - Dev error (by typing the wrong constant). Should be caught by any test that uses the libfunc.
  - Invalid Sierra type info. This should also be impossible at least without breaking changes and only between versions. If I'm not mistaken the type info is hardcoded into the Sierra type enums.

Reply:
I'm not sure I understand the problem then. If we remove `const N: usize` and always iterate `param_signatures`, then we would never encounter a SIGSEGV and we will always return the same number of values as we receive, making any dev error impossible. Maybe I'm missing something. Is `const N` used for performance reasons?

Reply:
@azteca1998 could you add a todo comment, reminding us to check if all noops return as many arguments as they receive?

Reply:
No need, they do not. Take the guarantee verify libfuncs, for example:
```cairo
extern fn u128_mul_guarantee_verify(guarantee: U128MulGuarantee) implicits(RangeCheck) nopanic;
```
Their signature is as follows:
```
(RangeCheck, U128MulGuarantee) -> (RangeCheck);
```

Reply:
Aah, that is true.